pkg/appconsts/versioned_consts.go

```diff
@@ -24,8 +24,13 @@ func SubtreeRootThreshold(_ uint64) int {
 // SquareSizeUpperBound is the maximum original square width possible
 // for a version of the state machine. The maximum is decided through
 // governance. See `DefaultGovMaxSquareSize`.
-func SquareSizeUpperBound(_ uint64) int {
-	return v1.SquareSizeUpperBound
+func SquareSizeUpperBound(v uint64) int {
+	switch v {
+	case testground.Version:
+		return testground.SquareSizeUpperBound
+	default:
+		return v1.SquareSizeUpperBound
+	}
```

Comment:
[question] Why doesn't this have case statements for each version?

```go
	case testground.Version:
		return testground.SquareSizeUpperBound
	case v1.Version:
		return v1.SquareSizeUpperBound
	case v2.Version:
		return v2.SquareSizeUpperBound
```

even if v1 and v2 have the same constants, this seems more readable and future proof in case we update the v2 constant prior to cutting the v2 release

Reply:
I actually disagree with this, as with each version, we'd have to go change each if statement across the entire codebase. besides being a bit annoying, there's no way to tell if we missed one or are being incosistent places. imo, just saying default reads as all versions use the same versions, which feels more concise then having to read each version and compare

Reply:
> I don't understand why silently defaulting to an old version is preferable over a panic.

there's a thread of two discussing this in the version PRs, so apologies if you are already aware of this point, but essentially were relying on there not being any human error to avoid a network halting bug. There's also not really any benefit, since the appversion MUST be checked when verifying the header, both by Comet and the light client. 

If the appversion is not expected or the node/LC doesn't know how to handle that version, it must halt. This means that there are no scenarios where the panic would serve a positive purpose. The only time that it would get hit is when we accidentally forget to update one of the ever increasing if statements, in which case the entire network halts for no reason, even if they upgraded. 

imo we _definitely_ should not introduce a panic. I'm also of the opinion that being specifying only the _changes_ for a constant is actually more readable. If we specify each value for each version and we want to find when something changed, we have to look up each constant and manually compare, which feels weird to me. That's more just a readability opinion, and won't halt the network, so I don't hold that strong of a view on that.

I think we've casually discussed this synchronously, but since this has come up multiple times, I'm happy to schedule a sync or open a discussion specifically for it if need.